src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.KeyValuePair.cs

```diff
@@ -424,15 +424,30 @@ private class ValueNameNullPolicy : JsonNamingPolicy
         [InlineData("{0")]
         [InlineData(@"{""Random"":")]
         [InlineData(@"{""Value"":1}")]
+        [InlineData(@"{null:1}")]
         [InlineData(@"{""Value"":1,2")]
         [InlineData(@"{""Value"":1,""Random"":")]
         [InlineData(@"{""Key"":1,""Key"":1}")]
+        [InlineData(@"{null:1,""Key"":1}")]
         [InlineData(@"{""Key"":1,""Key"":2}")]
         [InlineData(@"{""Value"":1,""Value"":1}")]
+        [InlineData(@"{""Value"":1,null:1}")]
         [InlineData(@"{""Value"":1,""Value"":2}")]
         public static void InvalidJsonFail(string json)
         {
             Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<KeyValuePair<int, int>>(json));
         }
+
+        [Fact]
+        public static void JsonPathIsAccurate()
+        {
+            string json = @"{""Key"":""1"",""Value"":2}";
+            JsonException ex = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<KeyValuePair<int, int>>(json));
+            Assert.Contains("$.Key", ex.ToString());
```

Comment:
```suggestion
            Assert.Contains("$.Key", ex.ToString());
            Assert.Equal("$.Key", ex.Path);
```

Would it be good to assert both?

Reply:
There are detailed tests for JSON path and `JsonException` are in `ExceptionTests` that provide this extra validation.